core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -28,60 +28,47 @@ var ExplorationEditorTranslationTab = function() {
   var translationWelcomeModal = element(
     by.css('.protractor-test-translation-tab-welcome-modal'));
 
-  this.exitTutorial = function() {
+  this.exitTutorial = async function() {
     // If the translation welcome modal shows up, exit it.
-    translationWelcomeModal.isPresent().then(function(isVisible) {
-      waitFor.visibilityOf(dismissWelcomeModalButton,
-        'Welcome modal not becoming visible').then(
-        () => {
-          waitFor.elementToBeClickable(
-            dismissWelcomeModalButton,
-            'Welcome modal is taking too long to appear');
-          dismissWelcomeModalButton.click();
-        },
-        (err) => {
-          // Since the welcome modal appears only once, the wait for its
-          // visibilty will only resolve once and timeout the other times.
-          // This is just an empty error function to catch the timeouts that
-          // happen when the the welcome modal has been dismissed once. If
-          // this is not present then protractor uses the default error
-          // function which is not appropriate in this case as this is not an
-          // error.
-        }
-      );
-    });
+    var isVisible = await translationWelcomeModal.isPresent();
+    if (isVisible) {
```

Comment:
How do we handle success/failure callbacks with await without .then()?

Reply:
We use try-catch I think!

Reply:
> This doesn't match the old behavior. We need to wait to see if the modal appears. If it times out, we do nothing and throw no errors. If it does appear, then we dismiss it.

Thanks for confirming the behaviour. I'll fix this one.

Reply:
I took a look. I think the behaviour has not changed with async/await because isPresent() returns a promise which resolves with a boolean value. Note that it doesn't have a reject callback:
https://github.com/angular/protractor/blob/e77885d7a60fd5a56bf624a5307fd41582f10bca/lib/element.ts#L444-L448

So if the if the element is visible it dismisses it and if not it continues execution.

Reply:
Sorry, I misinterpretted it. I just realised you were talking about `waitFor.visibilityOf(dismissWelcomeModalButton,..` part.

Reply:
appears fixed in current diff